content/en/docs/collector/howto-receiver-trace.md

```diff
@@ -0,0 +1,1912 @@
+
+>trace-receiver.go
+```go
+package tailtracer
+
+import (
+	"context"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+	
+)
+
+type tailtracerReceiver struct {
+    host         component.Host
+	cancel       context.CancelFunc
+	logger       *zap.Logger
+	nextConsumer consumer.Traces
+	config       *Config
+}
+
+func (tracemokRcvr *tailtracerReceiver) Start(ctx context.Context, host component.Host) error {
+    tracemokRcvr.host = host
+    ctx = context.Background()
+	ctx, tracemokRcvr.cancel = context.WithCancel(ctx)
+ 
+	interval, _ := time.ParseDuration(tracemokRcvr.config.Interval)
+	go func() {
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				tracemokRcvr.logger.Info("I should start processing traces now!")
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (tracemokRcvr *tailtracerReceiver) Shutdown(ctx context.Context) error {
+	tracemokRcvr.cancel()
+	return nil
+}
+
+```
+
+>### Reviewing the code
+>- Importing the `go.opentelemetry.io/collector/consumer` which is where the pipeline's consumer types and interfaces are declared.
+>- Importing the `go.uber.org/zap` package, which is what the Collector uses for it's logging capabilities.
+>- Added a `zap.Logger` field named `logger` so we can have access to the Collector's logger reference from within the receiver.
+>- Added a `consumer.Traces` field named `nextConsumer` so we can push the traces generated by the `tailtracer` receiver to the next consumer declared in the Collector's pipeline.
+>- Added a `Config` field named `config` so we can have access to receiver's configuration settings defined within the Collector's config.
+>- Added a variable named `interval` that will be initialized as a `time.Duration` based on the value of the `interval` settings of the `tailtracer` receiver defined within the Collector's config.  
+>- Added a `go func()` to implement the `ticker` mechanism so our receiver can generate traces every time the `ticker` reaches the amount of time specified by the `interval` variable and used the `tracemokRcvr.logger` field to generate a info message every time the receiver supposed to be generating traces.
+
+
+  
+
+The `tailtracerReceiver` type is now ready to be instantiated and keep all  meaningful information passed by it's factory.
+
+Open the `tailtracer/factory.go` file and navigate to the `createTracesReceiver()` function. 
+
+The receiver is only instantiated if it's declared as a component within a pipeline and the factory is responsible to make sure the next consumer (either a processor or exporter) in the pipeline is valid otherwise it should generate an error. 
+
+The Collector's api provides some standard error types to help the factory handle pipeline configurations. Your receiver factory should throw a `component.ErrNilNextConsumer` in case the next consumer has an issue and is passed as nil.
```

Comment:
```suggestion
The Collector's API provides some standard error types to help the factory handle pipeline configurations. Your receiver factory should throw a `component.ErrNilNextConsumer` in case the next consumer has an issue and is passed as nil.
```